torch/distributed/_composable/fsdp/_fsdp_collectives.py

```diff
@@ -173,8 +167,7 @@ def foreach_reduce_scatter(
                 fsdp_param.sharded_param.grad = new_sharded_dtensor_grad
             padded_sharded_numel = padded_unsharded_size.numel() // world_size
             flat_grad_offset += padded_sharded_numel
-        reduce_scatter_view_out_event = torch.cuda.Event()
-        reduce_scatter_view_out_event.record()
+        reduce_scatter_view_out_event = reduce_scatter_stream.record_event()
```

Comment:
Just for my understanding, with this semantic, the event don't need to be created under `with torch.cuda.stream(...)` right?

Reply:
Good question! I think you are right -- we do not need to create the event under the `with torch.cuda.stream()` context. `Stream.record_event()`: construct `event` -> `event.record(Stream)`, so it will record the stream the event in the calling stream (`reduce_scatter_stream` here).

Aside: You asking this made me realize that `event.record()` with no stream specified calls `torch.cuda.current_stream()`, so we anyway call `current_stream()`.